### PR TITLE
v2: improve package structure and replace mux with gin

### PR DESCRIPTION
**Pull request by waltzofpearls**
- Simpler subfolder and package structure
- Replaced Gorilla Mux with Gin
- Better Logger middleware with zap
- Improved graceful shutdown with cancelable context

### CODE REVIEW

**Comment by waltzofpearls** [comment]
- Simpler subfolder and package structure
- Replaced Gorilla Mux with Gin
- Better [Logger](./middleware/logger.go) middleware with zap
- Improved graceful shutdown with cancelable context

File: v2/examples/graphql/graph/status.graphqls [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# GraphQL schema example
+#
+# https://gqlgen.com/getting-started/
+
+type Status {
+  State: String!
```
Comment (terapaks) [inline v2/examples/graphql/graph/status.graphqls:6]: What is String!

**Review by chazlett — APPROVED** [review]
This looks great. Thanks for improving everything, but especially logging. I did it very naively before and this is much better.

**Review by terapaks — APPROVED** [review]

File: v2/examples/graphql/graph/status.graphqls (hunk 1 above)
Comment (terapaks) [reply v2/examples/graphql/graph/status.graphqls:6]: Response from Rollie:  this is graphql specific generated code.

Comment (waltzofpearls) [reply v2/examples/graphql/graph/status.graphqls:6]: Exclamation marks a field to be non-nullable. https://graphql.org/learn/schema/#object-types-and-fields

**Pull request merged**